### ISSUE TITLE
Setup Instructions for PROS plugin on existing Atom Install

### ISSUE THREAD

**Issue by KZeronimo** [purduesigbots/pros-atom3#40]
**This should be moved under `pros-atom3`**

#### Expected Behavior:
Currently using Atom and install pros plugin - expect the same experience as the PROS Editor

#### Actual Behavior:
There are a number of differences

1. Console behavior
2. Have to build and install cquery out of band
3. Maybe more

#### Steps to reproduce:
Install pros plugin on existing up to date install of Atom
Install PROS Editor
Note difference in behavior

#### System information:
Operating System (e.g. Windows, OS X, Ubuntu):
Windows 10

Atom Version:
Latest

Plugin Version (PROS version):
Latest

**Comment by edjubuh** [comment]
This sounds like a long term mission for us. I don't think we'll be interested in trying to package cquery with the pros-atom3 plugin. The console has been a great example of things going wrong when you expect native binaries to be included in the plugin. That being said, we do distribute pre-built cquery on [Azure](https://dev.azure.com/purdue-acm-sigbots/Installers/_build/results?buildId=516)

**Comment by HotelCalifornia** [comment]
we've definitely had a lot of issues with the console, but what differences in behavior do you mean?

**Comment by KZeronimo** [comment]
Thanks for the quick reply guys - let me rephrase my question

If I have Atom already installed and configured to taste and I want to use PROS - what is the process beyond installing the pros-atom3 package to get the same experience offered by the PROS Editor?

So far I know I need to

- Install the pros-atom3 package
- Build and install cquery
- What else?

I know I can run Atom and PROS Editor side by side and that might end up being the best answer - I don't know how much you've done in your fork of Atom to achieve the experience offered by the PROS Editor ...

**Comment by HotelCalifornia** [comment]
[The instructions for Linux](https://pros.cs.purdue.edu/v5/getting-started/linux.html#installing-the-editor) should get you what you need

**Comment by KZeronimo** [comment]
@HotelCalifornia - I've tried a variation of that by installing the pros-atom3 package - should I be installing the bootstrapper instead?

Just curious if you have tried that recipe and end up with Atom being at parity with PROS Editor

**Comment by HotelCalifornia** [comment]
Um yeah you should end up at parity with the editor by installing the bootstrapper, as that's basically what the editor does

**Comment by KZeronimo** [comment]
@HotelCalifornia - thank you - I'll try again fresh this time with the bootstrapper instead of the pros-atom3 package and let you know how it goes.

A point of clarification between the purpose of the bootstrapper and the pros-atom3 package would be helpful especially since the package catalog states - ... "You probably don't want to manually install this package." about the bootstrapper package.

**Comment by edjubuh** [comment]
You don't want to manually install `pros-boostrapper`. What went wrong with installation as outlined [here](https://github.com/purduesigbots/pros-atom3/issues/40#issuecomment-487080787)?

The end effect of `pros-boostrapper` and `pros-atom3` is going to be nearly identical. The bootstrapper has a package dependency on `pros-atom3`, `pros-atom3`'s package dependencies, and some minimap packages that we thought most users would appreciate being installed by default. If you just install `pros-atom3`, you should get all the necessary dependencies required for PROS to work inside Atom.

**Comment by KZeronimo** [comment]
@edjubuh - thanks for the reply - if I recall - when I perform a build action in the PROS Editor I get output to the terminal and I have a console being filtered for PROS output - the terminal interaction and console are missing in the Atom + PROS Package

Let me do this clean again and report back - I take it from out thread it should be as easy as the steps I outlined [here](https://github.com/purduesigbots/pros-atom3/issues/40#issuecomment-487080787) to achieve Atom + PROS = PROS Editor

**Comment by edjubuh** [comment]
> Let me do this clean again and report back - I take it from out thread it should be as easy as the steps I outlined here to achieve Atom + PROS = PROS Editor

I'll give a more thorough listing:
1. Install PROS CLI + Toolchain through either installers or manually with `pip` and from [Arm's website](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm)
2. Install Atom
3. Install cquery, either by building yourself or finding online
4. Install pros-atom3 in your Atom installation

**Comment by HotelCalifornia** [comment]
> You don't want to manually install `pros-boostrapper`.

To clarify, installing the bootstrapper from inside atom will install the old version of the atom package. You will need to use `apm` to install the correct version, as outlined in the instructions I linked above.
